app/src/main/java/com/example/mobile_development/MainActivity.kt

```diff
@@ -0,0 +1,244 @@
+import androidx.compose.material.icons.filled.VisibilityOff
+import androidx.compose.material3.Button
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import com.example.mobile_development.ui.theme.MobiledevelopmentTheme
+import java.util.regex.Pattern
+
+
+class MainActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            MobiledevelopmentTheme {
+                val navController =  rememberNavController()
+                // A surface container using the 'background' color from the theme
+                Surface(
+                    modifier = Modifier.fillMaxSize(),
+                    color = MaterialTheme.colorScheme.background
+                ) {
+                    Scaffold {
+                        NavHost(navController = navController,
+                            startDestination = "login",
+                        modifier = Modifier.padding(it)
+                        ) {
+                            composable("login") { LoginScreen(navigateToHomeScreen = { navController.navigate("home") }) }
+                            composable("home") { HomeScreen() }
+                        }
+                    }
+
+
+
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun LoginScreen(navigateToHomeScreen: () -> Unit={},modifier: Modifier = Modifier) {
+    var (email, setEmail) = remember { mutableStateOf("") }
+    var (password, setPassword) = remember { mutableStateOf("") }
+    var (showPass, setShowPass) = remember { mutableStateOf(false) }
+    var (isError, setIsError) = remember { mutableStateOf(false) }
+
+    fun ValidateEmail(): Boolean {
```

Comment:
Only composable functions are written in `PascalCase`. All others should be `camelCase`.